contracts/VaultV1.sol

```diff
@@ -72,20 +57,20 @@ contract VaultV1 is ERC4626, Ownable {
             "withdraw locked"
         );
 
-        // check how much underlying we have "on hand"
-        uint256 startUnderlying = UNDERLYING.balanceOf(address(this));
-
-        if (startUnderlying < amount && STRATEGY.withdrawable() >= amount) {
-            // not enough on hand but enough in the strategy. withdraw
-            STRATEGY.withdraw(amount);
-        } else if (startUnderlying < amount) {
-            // not enough on hand. Not enough in strategy. Revert to be safe.
+        // all funds are stored in strategy. See how much can be pulled
+        if (strategy.withdrawable() < amount) {
+            // we don't have enough funds
             revert("not enough funds in vault");
         }
 
+        // pull funds from strategy so they can be returned to the user
+        strategy.withdraw(amount);
+
         // update the users requested withdraw status
         requestedWithdraws[msg.sender] = 0;
         unlockTime[msg.sender] = 0;
+        // update global withdraw amount
+        totalRequestedWithdraws -= amount;
```

Comment:
If we rely on this logic but all requesters do not withdraw within the 24 hrs, does this still work as intended?
As the bot is relying on totalRequestedWithdraws to signal funds to be sent back

Reply:
Ideally the bot wouldn't rely on this. The bot should watch for events emitted from the strategy and return funds based on those. This param mainly exists as a fallback.

Its impossible to know how much needs to be returned given only one variable. You could decrement this when the bot returns funds however this then too closely couples (imo) the bot and vault. The bot should only really interact with the strategy and the vault should be agnostic.